flink-end-to-end-tests/test-scripts/common.sh

```diff
@@ -429,6 +429,7 @@ function check_logs_for_exceptions {
    | grep -v "org.apache.flink.runtime.JobException: Recovery is suppressed" \
    | grep -v "WARN  akka.remote.ReliableDeliverySupervisor" \
    | grep -v "RecipientUnreachableException" \
+   | grep -v "SerializedCheckpointException.unwrap" \
```

Comment:
Is it safe to filter out the exception? Can we be certain it is a false positive in all tests?

Reply:
WDYT about filtering out on a less generic part? Would it be possible to filter on e.g. `org.apache.flink.runtime.checkpoint.CheckpointException: Checkpoint was declined (task is closing)`?

I am afraid we will filter out exceptions we do not expect to happen.

Reply:
No, it doesn't work like that. 
For example, we have the log:
```
org.apache.flink.runtime.checkpoint.CheckpointException: Checkpoint was declined (task is closing)
        at org.apache.flink.runtime.messages.checkpoint.SerializedCheckpointException.unwrap(SerializedCheckpointException.java:51) ~[flink-dist_2.11-1.14-SNAPSHOT.jar:1.14-SNAPSHOT]
        at org.apache.flink.runtime.checkpoint.CheckpointCoordinator.receiveDeclineMessage(CheckpointCoordinator.java:988) ~[flink-dist_2.11-1.14-SNAPSHOT.jar:1.14-SNAPSHOT]
        at org.apache.flink.runtime.scheduler.ExecutionGraphHandler.lambda$declineCheckpoint$2(ExecutionGraphHandler.java:103) ~[flink-dist_2.11-1.14-SNAPSHOT.jar:1.14-SNAPSHOT]
        at org.apache.flink.runtime.scheduler.ExecutionGraphHandler.lambda$processCheckpointCoordinatorMessage$3(ExecutionGraphHandler.java:119) ~[flink-dist_2.11-1.14-SNAPSHOT.jar:1.14-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_322]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_322]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_322]
```
The existing filter already takes into account `CheckpointException` so after the filter will be applied we get:
```
at org.apache.flink.runtime.messages.checkpoint.SerializedCheckpointException.unwrap(SerializedCheckpointException.java:51)
```
But actually, we expect to have nothing after the filter is applied since this exception is expected. So right now we need to filter this line somehow. It is why I use `SerializedCheckpointException.unwrap`. I believe it is safe because we should pay attention to the first line of exception not a random line in the stacktrace. (in my example the important exception is `CheckpointException` but it is already in the white list)

Reply:
Ok, now I see. Thanks for the explanation!